From: Ole Streicher <olebole@debian.org>
Date: Thu, 29 Sep 2016 20:35:34 +0200
Subject: Build shared lib with SONAME and SOVERSION

---
 src/CMakeLists.txt | 31 ++++++++++++++++++++-----------
 1 file changed, 20 insertions(+), 11 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index eadc3a2..772fa89 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -232,18 +232,14 @@ set(PYTHON_FILES
 	python/functions.cpp)
 
 set(ALL_LIBRARIES
-	${BLAS_LIBRARIES} ${LAPACK_LIBRARIES}
 	${Boost_SYSTEM_LIBRARY}
-	${Boost_FILESYSTEM_LIBRARY} ${Boost_DATE_TIME_LIBRARY}
+	${Boost_DATE_TIME_LIBRARY}
 #	${Boost_PYTHON-PY36_LIBRARIES}
 	${Boost_PYTHON_LIBRARIES}
 	${Boost_SIGNALS_LIBRARY}
 	${FFTW3_LIB}
 	${CASACORE_LIBRARIES}
-	${LAPACK_lapack_LIBRARY}
-	${CFITSIO_LIBRARY}
 	${CMAKE_THREAD_LIBS_INIT}
-	${PNG_LIBRARIES} ${LIBXML2_LIBRARIES}
 	${PYTHON_LIBRARIES})
 
 if(GTKMM_FOUND)
@@ -263,11 +259,23 @@ endif(SIGCXX_FOUND)
 if(GTKMM_FOUND)
 	set(AOFLAGGER_PLOT_FILES ${PLOT_FILES})
 endif(GTKMM_FOUND)
-add_library(aoflagger SHARED ${AOFLAGGER_PLOT_FILES} ${IMAGING_FILES} ${INTERFACE_FILES} ${MSIO_FILES} ${QUALITY_FILES} ${STRATEGY_FILES} ${STRUCTURES_FILES} ${UTIL_FILES} ${PYTHON_FILES})
-set_target_properties(aoflagger PROPERTIES SOVERSION 0)
-target_link_libraries(aoflagger ${ALL_LIBRARIES})
-
-link_libraries(aoflagger)
+add_library(aoflagger-lib STATIC ${AOFLAGGER_PLOT_FILES} ${IMAGING_FILES} ${INTERFACE_FILES} ${MSIO_FILES} ${QUALITY_FILES} ${STRATEGY_FILES} ${STRUCTURES_FILES} ${UTIL_FILES} ${PYTHON_FILES})
+set_target_properties(aoflagger-lib PROPERTIES OUTPUT_NAME aoflagger)
+
+add_library(aoflagger-shared SHARED ${AOFLAGGER_PLOT_FILES} ${IMAGING_FILES} ${INTERFACE_FILES} ${MSIO_FILES} ${QUALITY_FILES} ${STRATEGY_FILES} ${STRUCTURES_FILES} ${UTIL_FILES} ${PYTHON_FILES})
+set_target_properties(aoflagger-shared PROPERTIES OUTPUT_NAME aoflagger)
+SET_TARGET_PROPERTIES(aoflagger-shared PROPERTIES SOVERSION 0)
+target_link_libraries(aoflagger-shared PRIVATE
+  ${Boost_THREAD_LIBRARY} ${Boost_SYSTEM_LIBRARY}
+  ${Boost_FILESYSTEM_LIBRARY} ${Boost_PYTHON_LIBRARIES}
+  ${FFTW3_LIB}
+  ${CASACORE_LIBRARIES}
+  ${CFITSIO_LIBRARY}
+  ${CMAKE_THREAD_LIBS_INIT}
+  ${PNG_LIBRARIES} ${LIBXML2_LIBRARIES}
+  ${CMAKE_THREAD_LIBS_INIT})
+
+link_libraries(aoflagger-shared)
 
 add_library(python_aoflagger SHARED python/pyinterface.cpp)
 set_target_properties(python_aoflagger PROPERTIES SOVERSION 0)
@@ -328,5 +336,6 @@ if(BOOST_ASIO_H_FOUND AND SIGCXX_FOUND AND GTKMM_FOUND)
 endif(BOOST_ASIO_H_FOUND AND SIGCXX_FOUND AND GTKMM_FOUND)
 
 install (TARGETS aoflagger-bin DESTINATION bin)
-install (TARGETS aoflagger DESTINATION lib) 
+install (TARGETS aoflagger-shared DESTINATION lib/${DEB_BUILD_MULTIARCH}) 
+install (TARGETS aoflagger-lib DESTINATION lib/${DEB_BUILD_MULTIARCH}) 
 install (FILES interface/aoflagger.h DESTINATION include)
