From: Ole Streicher <olebole@debian.org>
Date: Thu, 29 Sep 2016 20:35:34 +0200
Subject: Build shared lib with SONAME and SOVERSION

---
 src/CMakeLists.txt | 30 +++++++++++++++++++-----------
 1 file changed, 19 insertions(+), 11 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index cdc5984..f8f4aa9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -223,15 +223,10 @@ else()
 endif(BOOST_ASIO_H_FOUND AND SIGCXX_FOUND)
 
 set(ALL_LIBRARIES
-	${BLAS_LIBRARIES} ${LAPACK_LIBRARIES}
 	${Boost_THREAD_LIBRARY} ${Boost_SYSTEM_LIBRARY}
-	${Boost_FILESYSTEM_LIBRARY} ${Boost_DATE_TIME_LIBRARY} ${Boost_SIGNALS_LIBRARY}
+	${Boost_DATE_TIME_LIBRARY}
 	${FFTW3_LIB}
 	${CASACORE_LIBRARIES}
-	${LAPACK_lapack_LIBRARY}
-	${CFITSIO_LIBRARY}
-	${CMAKE_THREAD_LIBS_INIT}
-	${PNG_LIBRARIES} ${LIBXML2_LIBRARIES}
 	${CMAKE_THREAD_LIBS_INIT})
 
 if(GTKMM_FOUND AND GLIBMM_FOUND)
@@ -248,11 +243,23 @@ if(SIGCXX_FOUND)
   set(ALL_LIBRARIES ${ALL_LIBRARIES} ${SIGCXX_LIBRARIES})
 endif(SIGCXX_FOUND)
 
-add_library(aoflagger SHARED ${IMAGING_FILES} ${INTERFACE_FILES} ${MSIO_FILES} ${QUALITY_FILES} ${STRATEGY_FILES} ${STRUCTURES_FILES} ${UTIL_FILES})
-SET_TARGET_PROPERTIES(aoflagger PROPERTIES SOVERSION 0)
-target_link_libraries(aoflagger ${ALL_LIBRARIES})
+add_library(aoflagger-lib STATIC ${IMAGING_FILES} ${INTERFACE_FILES} ${MSIO_FILES} ${QUALITY_FILES} ${STRATEGY_FILES} ${STRUCTURES_FILES} ${UTIL_FILES})
+set_target_properties(aoflagger-lib PROPERTIES OUTPUT_NAME aoflagger)
 
-link_libraries(aoflagger)
+add_library(aoflagger-shared SHARED ${IMAGING_FILES} ${INTERFACE_FILES} ${MSIO_FILES} ${QUALITY_FILES} ${STRATEGY_FILES} ${STRUCTURES_FILES} ${UTIL_FILES})
+set_target_properties(aoflagger-shared PROPERTIES OUTPUT_NAME aoflagger)
+SET_TARGET_PROPERTIES(aoflagger-shared PROPERTIES SOVERSION 0)
+target_link_libraries(aoflagger-shared PRIVATE
+  ${Boost_THREAD_LIBRARY} ${Boost_SYSTEM_LIBRARY}
+  ${Boost_FILESYSTEM_LIBRARY}
+  ${FFTW3_LIB}
+  ${CASACORE_LIBRARIES}
+  ${CFITSIO_LIBRARY}
+  ${CMAKE_THREAD_LIBS_INIT}
+  ${PNG_LIBRARIES} ${LIBXML2_LIBRARIES}
+  ${CMAKE_THREAD_LIBS_INIT})
+
+link_libraries(aoflagger-shared)
 
 add_library(aoflaggerremote OBJECT ${REMOTE_FILES})
 set(AOFLAGGERREMOTE_OBJECT $<TARGET_OBJECTS:aoflaggerremote>)
@@ -307,5 +314,6 @@ if(BOOST_ASIO_H_FOUND AND SIGCXX_FOUND AND GTKMM_FOUND)
 	install (TARGETS aoquality aoremoteclient DESTINATION bin)
 endif(BOOST_ASIO_H_FOUND AND SIGCXX_FOUND AND GTKMM_FOUND)
 
-install (TARGETS aoflagger DESTINATION lib) 
+install (TARGETS aoflagger-shared DESTINATION lib/${DEB_BUILD_MULTIARCH}) 
+install (TARGETS aoflagger-lib DESTINATION lib/${DEB_BUILD_MULTIARCH}) 
 install (FILES interface/aoflagger.h DESTINATION include)
