From: Gijs Molenaar <gijs@pythonic.nl>
Date: Mon, 19 Mar 2018 12:49:10 +0000
Subject: Declare-pow10

---
 src/quality/loghistogram.h | 3 ---
 src/util/numberparser.h    | 3 ---
 2 files changed, 6 deletions(-)

diff --git a/src/quality/loghistogram.h b/src/quality/loghistogram.h
index 1150786..6b62ee1 100644
--- a/src/quality/loghistogram.h
+++ b/src/quality/loghistogram.h
@@ -10,10 +10,7 @@
 
 #include "../util/serializable.h"
 
-//# pow10 seems to be undefined on OS-X
-#ifdef __APPLE__
 # define pow10(x) pow(10., (x))
-#endif
 
 
 class LogHistogram : public Serializable
diff --git a/src/util/numberparser.h b/src/util/numberparser.h
index 509ae4f..889cba0 100644
--- a/src/util/numberparser.h
+++ b/src/util/numberparser.h
@@ -2,10 +2,7 @@
 
 #include <cmath>
 
-//# pow10 seems to be undefined on OS-X
-#ifdef __APPLE__
 # define pow10(x) pow(10., (x))
-#endif
 
 
 class NumberParsingException : public std::runtime_error
