From: Ole Streicher <olebole@debian.org>
Date: Wed, 19 Aug 2020 14:21:47 +0200
Subject: Disable test cases that don't work before installation

---
 test/interface/interfacetest.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/interface/interfacetest.cpp b/test/interface/interfacetest.cpp
index 2d2555e..50fa2e8 100644
--- a/test/interface/interfacetest.cpp
+++ b/test/interface/interfacetest.cpp
@@ -8,11 +8,13 @@
 
 BOOST_AUTO_TEST_SUITE(interface, *boost::unit_test::label("interface"))
 
+#if 0
 BOOST_AUTO_TEST_CASE(find_strategy) {
   aoflagger::AOFlagger flagger;
   std::string defaultStrategy = flagger.FindStrategyFile();
   BOOST_CHECK_NE(defaultStrategy, "");
 }
+#endif
 
 BOOST_AUTO_TEST_CASE(load_nonexisting_strategy) {
   aoflagger::AOFlagger flagger;
@@ -31,6 +33,7 @@ BOOST_AUTO_TEST_CASE(make_image_set) {
   BOOST_CHECK_GE(imageSet.HorizontalStride(), 100);
 }
 
+#if 0
 BOOST_AUTO_TEST_CASE(run_default_strategy_without_input) {
   aoflagger::AOFlagger flagger;
   aoflagger::Strategy strategy =
@@ -93,6 +96,7 @@ BOOST_AUTO_TEST_CASE(runs) {
   BOOST_CHECK_GT(mask2DB.GetCount<true>(), 0);
   BOOST_CHECK_LT(mask2DB.GetCount<true>(), width * height / 5);
 }
+#endif
 
 BOOST_AUTO_TEST_CASE(version) {
   short major, minor, subminor;
