From: Ole Streicher <olebole@debian.org>
Date: Fri, 28 Feb 2020 14:38:56 +0100
Subject: Install libaoflagger.so into multi-arch path

Closes: #962390
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d625317..cd5c543 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -279,7 +279,7 @@ add_library(aoflagger-lib SHARED ${ALLFILES})
 set_target_properties(aoflagger-lib PROPERTIES SOVERSION 0)
 set_target_properties(aoflagger-lib PROPERTIES OUTPUT_NAME aoflagger)
 target_link_libraries(aoflagger-lib ${ALL_LIBRARIES})
-install (TARGETS aoflagger-lib DESTINATION lib) 
+install (TARGETS aoflagger-lib DESTINATION lib/${DEB_BUILD_MULTIARCH}) 
 
 add_executable(aoquality applications/aoquality.cpp)
 target_link_libraries(aoquality aoflagger-lib ${ALL_LIBRARIES})
