From: Ole Streicher <olebole@debian.org>
Date: Mon, 30 Aug 2021 11:05:37 +0200
Subject: Use system provided pybind11

---
 CMakeLists.txt | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 86146fa..3adc2f5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,11 +30,6 @@ if(GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
     endif()
 endif()

-# Include aocommon/pybind11 headers
-include_directories("${CMAKE_SOURCE_DIR}/aocommon/include")
-add_subdirectory("${CMAKE_SOURCE_DIR}/external/pybind11")
-include_directories(${pybind11_INCLUDE_DIR})
-
 set(CASACORE_MAKE_REQUIRED_EXTERNALS_OPTIONAL TRUE)
 find_package(Casacore REQUIRED COMPONENTS casa ms tables measures)
 include_directories(${CASACORE_INCLUDE_DIRS})
@@ -61,6 +56,11 @@ find_package(PythonInterp REQUIRED)
 message(STATUS "Using python version ${PYTHON_VERSION_STRING}")
 include_directories(${PYTHON_INCLUDE_DIRS})

+# Include pybind11 headers
+find_package(pybind11 REQUIRED)
+include_directories("${CMAKE_SOURCE_DIR}/aocommon/include")
+include_directories(${pybind11_INCLUDE_DIR})
+
 # boost::alignment requires Boost 1.56
 find_package(Boost 1.56.0 REQUIRED COMPONENTS date_time filesystem system unit_test_framework)
 include_directories(${Boost_INCLUDE_DIR})
